Automattic/wp-calypso
# Subscriptions management: Add 'Subscription not found' state

![Screen Shot 2023-05-25 at 16 17 35](https://github.com/Automattic/wp-calypso/assets/2019970/48222b87-e671-434b-bb2c-488b5aad2fe7)


Closes https://github.com/Automattic/wp-calypso/issues/77305

## Proposed Changes

* Display the error notice when fetching a site subscription details is unsuccessful
* Handle both type of response errors
    1. Where the server itself is reporting an error. This is the classical HTTP error response.
    2. Where the endpoint is reporting an error. This is an error that is wrapped in a payload of 200 "OK" response.

## Testing Instructions


1. Test the classical HTTP error response:
    * Go to http://calypso.localhost:3000/subscriptions/site/a to test for the classical HTTP error response
    * Ensure the error message is displayed

2. Test the classical HTTP error response:
    * Go to http://calypso.localhost:3000/subscriptions/site/$site-id, where the `$site-id` is a valid ID of a site that you are not subscribed to (it can be an existing site, or not, but it is important that it matches blog ID regex on the server side), e.g. https://wordpress.com/subscriptions/site/12345678.
    * Ensure the error message is displayed

3. Test for no regressions
    * Go to http://calypso.localhost:3000/subscriptions/site/$site-id, where the `$site-id` is a valid ID of a site that you  subscribed to
    * Ensure the error message is not displayed 

## Pre-merge Checklist
